### PR TITLE
Fix lollipop when canonical transcript is missing

### PR DESCRIPTION
**Pull request by inodb**
Get all transcripts with given hugo symbol and show longest.

Fix https://github.com/cBioPortal/cbioportal/issues/3922

### CODE REVIEW

File: src/pages/resultsView/mutation/MutationMapperStore.ts [hunk 1]
```diff
@@ -142,10 +143,33 @@ export class MutationMapperStore {
...
+    }, undefined);
+
     readonly canonicalTranscript = remoteData<EnsemblTranscript | undefined>({
+        await: () => [
+            this.allTranscripts
+        ],
         invoke: async()=>{
             if (this.gene) {
-                return fetchCanonicalTranscript(this.gene.hugoGeneSymbol, this.isoformOverrideSource);
+                return fetchCanonicalTranscript(this.gene.hugoGeneSymbol, this.isoformOverrideSource).catch(async() => {
```
Comment (onursumer) [inline src/pages/resultsView/mutation/MutationMapperStore.ts]: This seems to me a bit complicated as a return statement, maybe we should move the logic within the catch block into another utility function?

File: src/pages/resultsView/mutation/MutationMapperStore.ts [hunk 2]
```diff
@@ -142,10 +143,33 @@ export class MutationMapperStore {
...
+                    console.log("No canonical transcript for: " + this.gene.hugoGeneSymbol);
+                    const transcript = _.maxBy(this.allTranscripts.result, (t:EnsemblTranscript) => t.proteinLength);
+                    if (transcript) {
+                        console.log("Using transcript with longest length instead: " + transcript);
+                        return transcript;
+                    }
+                })
```
Comment (onursumer) [inline src/pages/resultsView/mutation/MutationMapperStore.ts]: missing semicolon!

File: src/pages/resultsView/mutation/MutationMapperStore.ts [hunk 3]
```diff
@@ -142,10 +143,33 @@ export class MutationMapperStore {
...
+            this.allTranscripts
+        ],
         invoke: async()=>{
             if (this.gene) {
-                return fetchCanonicalTranscript(this.gene.hugoGeneSymbol, this.isoformOverrideSource);
+                return fetchCanonicalTranscript(this.gene.hugoGeneSymbol, this.isoformOverrideSource).catch(async() => {
+                    console.log("No canonical transcript for: " + this.gene.hugoGeneSymbol);
+                    const transcript = _.maxBy(this.allTranscripts.result, (t:EnsemblTranscript) => t.proteinLength);
+                    if (transcript) {
+                        console.log("Using transcript with longest length instead: " + transcript);
```
Comment (onursumer) [inline src/pages/resultsView/mutation/MutationMapperStore.ts]: I have always had mixed feelings about `console.log` :)

File: src/pages/resultsView/mutation/MutationMapperStore.ts (hunk 1 above)
Comment (alisman) [reply src/pages/resultsView/mutation/MutationMapperStore.ts]: is it a failure that we expect or a real break down?

Comment (inodb) [reply src/pages/resultsView/mutation/MutationMapperStore.ts]: one that we expect. As a backup it uses the same service, so a real breakdown would still fail

File: src/pages/resultsView/mutation/MutationMapperStore.ts (hunk 3 above)
Comment (inodb) [reply src/pages/resultsView/mutation/MutationMapperStore.ts]: same, if we have a better way of logging warnings would be good

**Comment by inodb** [comment]
Changed this to master, since current bug is there

**Pull request merged**